transcom/mymove
# B 19181 create pptas api

Linked BLI: [B-19181](https://www13.v1host.com/USTRANSCOM38/TeamRoom.mvc/Show/848240)
Linked Integration PR: [B 19181 int create pptas api](https://github.com/transcom/mymove/pull/12629)
##How to test:

#Testing Prerequisites

1. Postman is required to test. 
2. PPTAS is acessed through TLS similar to acessing Prime API. In setup of Postman, you must add the mTLS certificates to postman. See [Setup Postman to make mTLS calls](https://transcom.github.io/mymove-docs/docs/tools/postman/setup-postman-to-make-mutual-tls-api-calls). This will cover your authentication to the API.
3. Easiest way to get the API definition into postman is to import it as a Postman collection with OpenAI Spec 2.0 using the generated pptas.yaml file under /swagger.
4. Setup an environment in postman for making Prime API calls, then set an env variable for the base url and api (Optional, but otherwise you will have to manually add base url to the request). ![EnvSetup](https://github.com/transcom/mymove/assets/150186792/b4665dc5-ecfa-4cfd-819a-40ebff057179)
5. Setup a collection variable for the path routing from base url to api path (`prime/pptas`). Set apiPath under collection variables as shown here: 
![CollectionVariableSettings](https://github.com/transcom/mymove/assets/150186792/d676e83e-1942-41f7-bdda-b2d2f21372d6)
 6. Click on listMoves to edit the request to the following: `{{baseUrl}}/{{apiPath}}/moves?since=<dateTime>`
![RouteSetup](https://github.com/transcom/mymove/assets/150186792/cbff0f3d-73a2-44d2-bb04-a74e6bbc72ab)
7, Both client and server must be running.

#How To Test

1. Go to [PPTAS Swagger](http://primelocal:3000/swagger-ui/pptas.html) and verify that the page loads and there is a single endpoint here. 
![PPTAS SwaggerScreenshot](https://github.com/transcom/mymove/assets/150186792/77e4b048-198b-4204-8ba5-78d524575e53)

2. Go under that endpoint and try calling it using the example date (or any date). 
![PPTAS moves](https://github.com/transcom/mymove/assets/150186792/d56a0292-803f-40b7-8dd2-50ddadd7af73)

3. The page should show the JSON results under a 200 result.
4. Within postman run the request using the one you setup in the pre requisites earlier. This should return a 200 result the reuslts in the response body. 
![PPTASRequestPostman](https://github.com/transcom/mymove/assets/150186792/6fab5aea-9980-4e78-a1ac-5f2cf89cf220)
